ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -719,14 +738,27 @@ private boolean verifyJoin(
               rightExpression
           ));
         }
+      } else {
+        // primary key join detected
+
+        final SqlType leftKeyType = Iterables.getOnlyElement(leftNode.getSchema().key()).type();
+        final SqlType rightKeyType = Iterables.getOnlyElement(rightNode.getSchema().key()).type();
+
+        verifyJoinConditionTypes(
```

Comment:
nit: if we're adding this check for regular table-table joins in this PR as well, can we also add the check for stream-stream and stream-table joins here as well for consistency? Or at least file an issue so we don't forget about it and leave the inconsistency.

Reply:
Let's file a ticket. For stream-stream and stream-table is more complex as we support expression...

Reply:
Actually, we should do this when adding expression support for FK-joins. Added a comment to the exiting ticket.